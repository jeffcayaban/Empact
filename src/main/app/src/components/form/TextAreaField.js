import React from 'react'
import { FormGroup, ControlLabel, FormControl, HelpBlock } from 'react-bootstrap';

/**
 * Is a generic text area field with validation to identify whether the field is empty.
 */

class TextAreaField extends React.PureComponent {

    render() {
        const { id, label, value, maxLength, validationState, ...props } = this.props;
        const currentFieldLength = value !== undefined ? value.length : 0;

        return (
            <div>
                <FormGroup controlId={id} validationState={validationState}>
                    <ControlLabel>{label}</ControlLabel>
                    <FormControl className={"petitionTextAreaField"} componentClass={"textarea"} value={value}
                                 maxLength={maxLength} {...props} />
                    <FormControl.Feedback />
                    <div className={"nonEmptyFieldWarningContainer"}>
                        <div className={"nonEmptyFieldWarning"}>
                            { validationState === 'error' && <HelpBlock>This field cannot be empty.</HelpBlock> }
                        </div>
                        <div className={"titleLengthCounter"}>
                            <p>{currentFieldLength}/{maxLength}</p>
                        </div>
                    </div>
                </FormGroup>
            </div>
        );
    }
}

export default TextAreaField;