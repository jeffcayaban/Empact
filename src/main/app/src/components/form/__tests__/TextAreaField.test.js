import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import TextAreaField from "../TextAreaField";

describe('TextAreaField', () => {

    it('should render successfully', () => {
        const component = shallow(<TextAreaField id={"testId"} label={"testLabel"} maxLength={50} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully', () => {
        const component = shallow(<TextAreaField
            id={"testId"}
            label={"testLabel"}
            maxLength={50}
            validationState={"error"}
        />);
        expect(toJson(component)).toMatchSnapshot();
    });
});