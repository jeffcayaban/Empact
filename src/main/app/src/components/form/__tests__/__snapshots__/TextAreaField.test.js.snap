// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaField should render successfully 1`] = `
<div>
  <FormGroup
    bsClass="form-group"
    controlId="testId"
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      testLabel
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      className="petitionTextAreaField"
      componentClass="textarea"
      maxLength={50}
    />
    <FormControlFeedback
      bsClass="form-control-feedback"
      bsRole="feedback"
    />
    <div
      className="nonEmptyFieldWarningContainer"
    >
      <div
        className="nonEmptyFieldWarning"
      />
      <div
        className="titleLengthCounter"
      >
        <p>
          0
          /
          50
        </p>
      </div>
    </div>
  </FormGroup>
</div>
`;

exports[`TextAreaField should render successfully 2`] = `
<div>
  <FormGroup
    bsClass="form-group"
    controlId="testId"
    validationState="error"
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      testLabel
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      className="petitionTextAreaField"
      componentClass="textarea"
      maxLength={50}
    />
    <FormControlFeedback
      bsClass="form-control-feedback"
      bsRole="feedback"
    />
    <div
      className="nonEmptyFieldWarningContainer"
    >
      <div
        className="nonEmptyFieldWarning"
      >
        <HelpBlock
          bsClass="help-block"
        >
          This field cannot be empty.
        </HelpBlock>
      </div>
      <div
        className="titleLengthCounter"
      >
        <p>
          0
          /
          50
        </p>
      </div>
    </div>
  </FormGroup>
</div>
`;
